lib/std/jsbigints.nim

```diff
@@ -0,0 +1,159 @@
+## Arbitrary precision integers.
+## * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
+when not defined(js) and not defined(nimdoc):
+  {.fatal: "Module jsbigints is designed to be used with the JavaScript backend.".}
+
+type JsBigInt* = ref object of JsRoot ## Arbitrary precision integer for JavaScript target.
+
+func newJsBigInt*(integer: SomeInteger): JsBigInt {.importjs: "BigInt(#)".} =
+  ## Constructor for `JsBigInt`.
+  runnableExamples:
+    doAssert newJsBigInt(1234567890) == newJsBigInt"1234567890"
+
+func newJsBigInt*(integer: cstring): JsBigInt {.importjs: "BigInt(#)".} =
+  ## Constructor for `JsBigInt`.
+  runnableExamples:
+    doAssert newJsBigInt"-1" == newJsBigInt"1" - newJsBigInt"2"
+
+func toCstring*(this: JsBigInt; radix: int): cstring {.importjs: "#.toString(#)".} =
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+  runnableExamples:
+    doAssert newJsBigInt"2147483647".toCstring(2) == "1111111111111111111111111111111".cstring
+
+func toCstring*(this: JsBigInt): cstring {.importjs: "#.toString()".} # asserted on $
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func `$`*(this: JsBigInt): string =
+  ## Return a string representation of `JsBigInt`.
+  runnableExamples: doAssert $newJsBigInt"1024" == "1024"
+  $toCstring(this)
+
+func toInt*(bits: int; a: JsBigInt): JsBigInt {.importjs: "BigInt.asIntN(#, #)".} =
+  ## Wrap `a` to a signed `JsBigInt` of `bits` bits, ie between `-2 ^ (bits - 1)` and `2 ^ (bits - 1) - 1`.
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asIntN
+  runnableExamples:
+    doAssert toInt(32, newJsBigInt"2147483647") == newJsBigInt"2147483647"
+
+func toUint*(bits: int; a: JsBigInt): JsBigInt {.importjs: "BigInt.asUintN(#, #)".} =
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asUintN
+  runnableExamples:
+    doAssert toUint(32, newJsBigInt"2147483647") == newJsBigInt"2147483647"
+
+func toInt*(a: JsBigInt): int {.importjs: "Number(#)".} =
```

Comment:
rename `toNumber` given that this doesn't do any bounds check; 

- [ ] in future work we can add a `toSafeInt` which does bounds check to return a safe interger (refs https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger)